octopoes/octopoes/repositories/scan_profile_repository.py

```diff
@@ -41,6 +41,7 @@ def get_bulk(self, references: set[Reference], valid_time: datetime) -> list[Sca
 class XTDBScanProfileRepository(ScanProfileRepository):
     object_type = "ScanProfile"
     pk_prefix = "xt/id"
+    typeadapter = TypeAdapter(ScanProfile)
```

Comment:
![image](https://github.com/user-attachments/assets/52623e77-708d-430f-aa5b-18dac633e706)

Reply:
This reduces the runtime for a loop that does no changes by about 30%

Reply:
Nice catch! Reusing the adapter will indeed prevent recreation of schemas. Unfortunately we do this on other places as well, I'll pick them up in a following PR.

Regarding the line, it's probably better to rename it to `scan_profile_adapter` and move it out of the class, either in the same module or in the `octopoes.models` module where the different scan profile types are declared. This way it can also be reused.

Reply:
We could add a more generic Dict / lookup table of the various models? We could even create a caching proxy/factory for all these adapters which only instantiates them if they are used.